### PR TITLE
form-dropdown react WIP

### PR DESCRIPTION
**Pull request by alexanderzobnin**
React replacement for `<gf-form-dropdown>`. Work in progress.

### CODE REVIEW

File: public/app/core/components/form_dropdown/FormDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+function convertToOption(value) {
+  let option = {
+    value: value
+  };
+  option[LABEL_KEY] = value;
+  return option;
+}
+
+react2AngularDirective('gfFormDropdownReact', FormDropdown, [
```
Comment (torkelo) [inline public/app/core/components/form_dropdown/FormDropdown.tsx]: move this to https://github.com/grafana/grafana/blob/master/public/app/core/angular_wrappers.ts  

trying to make the react component pure react so when we use them from other react components we do not need to depend on angular

File: public/app/core/components/form_dropdown/FormDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    }
+  }
+
+  getValueLabel() {
+    return this.selectControl.find('.Select-value-label');
+  }
+
+  onOpen() {
+    // Clear input label
+    this.getValueLabel().text("");
```
Comment (torkelo) [inline public/app/core/components/form_dropdown/FormDropdown.tsx]: I really like that the current value is still visible when open it so think we should remove this.

File: public/app/core/components/form_dropdown/FormDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+// import ReactDOM from 'react-dom';
+import Select from 'react-select';
+import { AsyncCreatable } from 'react-select';
+import _ from 'lodash';
+import $ from 'jquery';
+import 'react-select/scss/default.scss';
```
Comment (torkelo) [inline public/app/core/components/form_dropdown/FormDropdown.tsx]: move this import and the changes in _gf_form.scss to a new component file named form_dropdown.scss

**Comment by torkelo** [comment]
this is looking good!  Will be restyled a bit in develop branch to look better with the new form styles

**Comment by bergquist** [comment]
@alexanderzobnin whats the status here?

**Comment by bergquist** [comment]
@alexanderzobnin bump

**Comment by alexanderzobnin** [comment]
I should check the state of this. It wasn't a high-priority issue anyway.

**Comment by torkelo** [comment]
closing this for now as it was not working in a dynamic width setup, and we can revisit this when we have a clear use case

**Pull request closed without merging**